src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -0,0 +1,171 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DO_BEFORE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EXPECTED_TIME;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.assignment.Assignment;
+import seedu.address.model.assignment.Deadline;
+import seedu.address.model.assignment.ModuleCode;
+import seedu.address.model.assignment.Name;
+import seedu.address.model.assignment.Remind;
+import seedu.address.model.assignment.Schedule;
+import seedu.address.model.assignment.Task;
+import seedu.address.model.lesson.Lesson;
+
+/**
+ * Sets reminders for an assignment identified using it's displayed index from the address book.
+ */
+public class ScheduleCommand extends Command {
+
+    public static final String START_TIME = "06:00";
+    public static final String END_TIME = "23:59";
+    public static final int MAX_HOURS = 5;
+    public static final int MIN_HOURS = 1;
+    public static final String COMMAND_WORD = "schedule";
+    public static final String MESSAGE_SCHEDULE_ASSIGNMENT_SUCCESS = "Schedule Assignment: %1$s";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Schedule the assignment identified by the index number used in the displayed assignment list. "
+            + "The suggested time will be within working hours from " + START_TIME + " to " + END_TIME + "\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_EXPECTED_TIME + "EXPECTED HOURS (must be between " + MIN_HOURS + " and " + MAX_HOURS + " hours) "
+            + PREFIX_DO_BEFORE + "BEFORE "
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_EXPECTED_TIME + "2 " + PREFIX_DO_BEFORE + "23-12-2020 2359";
+
+    private static final LocalTime WORKING_START_TIME = LocalTime.parse(START_TIME, DateTimeFormatter.ISO_TIME);
+    private static final LocalTime WORKING_END_TIME = LocalTime.parse(END_TIME, DateTimeFormatter.ISO_TIME);
+
+    private final Index targetIndex;
+    private final Deadline doBefore;
+    private final int expectedTime;
```

Comment:
maybe can rename to expectedHours since it's referring to hours

Reply:
I was thinking of leaving it as Time so we can decide more on can it be minutes, days. But I have changed it to hours!